modules/pydamage/main.nf

```diff
@@ -0,0 +1,46 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process PYDAMAGE {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+        conda (params.enable_conda ? "bioconda::pydamage=0.61" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/pydamage:0.62--pyhdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/pydamage:0.62--pyhdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(bam), path(bai)
+
+    output:
+    tuple val(meta), path("pydamage_results/pydamage_results.csv"), path("pydamage_results/pydamage_filtered_results.csv"), emit: csv
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    pydamage \\
+        analyze \\
+        $options.args \\
+        -p $task.cpus \\
+        $bam
+
+    pydamage \\
+        filter \\
+        pydamage_results/pydamage_results.csv
+
+
+    pydamage --version | sed -e 's/pydamage, version //g' > ${software}.version.txt
```

Comment:
We apparently have to wrap the `<tool> --version` in a `echo \$(tool -version)` a lá: https://github.com/nf-core/modules/blob/6ff995e93ddec11d8c3112edcee231ce6777a15d/modules/bowtie2/align/main.nf#L48

Reply:
changed